### PR TITLE
[ISSUE #4345] Add support for polaris config sync

### PR DESCRIPTION
**Pull request by fabian4**
Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

---
## TODO:

Related issue: https://github.com/apache/shenyu/issues/4345

### ShenYu admin
- [x] Support for loading configurations from Polaris.
- [x] ~~Use [polaris-sync](https://github.com/polarismesh/polaris-sync) to sync config.~~
- [x] Update [polaris-java](https://github.com/polarismesh/polaris-java) https://github.com/polarismesh/polaris-java/pull/301
- [x] Support for pushing configurations to Polaris.https://github.com/polarismesh/polaris-java/issues/157
- [x] ~~https://github.com/polarismesh/polaris-java/pull/341~~
- [x] https://github.com/polarismesh/polaris/issues/1134
- [x] Add relevant test case.

### ShenYu bootstarap
- [x] Add PolarisListener for config changing.
- [x] Put it in a polaris stater.
- [x] Add relevant test case.

### CODE REVIEW

**Comment by fabian4** [comment]
As for support pushing configuration to Polaris:

The polaris client does not contain a method to publish the config to Polaris.

> See as: https://github.com/polarismesh/polaris-java/blob/main/polaris-configuration/polaris-configuration-api/src/main/java/com/tencent/polaris/configuration/api/core/ConfigFileService.java

```java
package org.apache.shenyu.admin.listener.polaris;

@Override
public void publishConfig(String dataId, Object data) {
    LOG.warn("Config upload not support yet, please upload it in polaris first");
}
```

So if we change on the admin dashboard only, it can not sync to Polaris and the bootstrap application.

**Comment by damonxue** [comment]
> As for support pushing configuration to Polaris:
> 
> The polaris client does not contain a method to publish the config to Polaris.
> 
> > See as: https://github.com/polarismesh/polaris-java/blob/main/polaris-configuration/polaris-configuration-api/src/main/java/com/tencent/polaris/configuration/api/core/ConfigFileService.java
> 
> ```java
> package org.apache.shenyu.admin.listener.polaris;
> 
> @Override
> public void publishConfig(String dataId, Object data) {
>     LOG.warn("Config upload not support yet, please upload it in polaris first");
> }
> ```
> 
> So if we change on the admin dashboard only, it can not sync to Polaris and the bootstrap application.

Now, Polaris-java SDK hasn't supported the sync config. But there is the other way to sync.
Refer to [config sync codes](https://github.com/polarismesh/polaris-sync/blob/main/polaris-sync-config-plugins/config-polaris/src/main/java/cn/polarismesh/polaris/sync/config/plugins/polaris/PolarisRestUtils.java)

**Comment by damonxue** [comment]
Pls add my WeChat(id: Fibonacci_stack), there is a group to support Polaris.

**Comment by fabian4** [comment]
> > As for support pushing configuration to Polaris:
> > The polaris client does not contain a method to publish the config to Polaris.
> > > See as: https://github.com/polarismesh/polaris-java/blob/main/polaris-configuration/polaris-configuration-api/src/main/java/com/tencent/polaris/configuration/api/core/ConfigFileService.java
> > 
> > 
> > ```java
> > package org.apache.shenyu.admin.listener.polaris;
> > 
> > @Override
> > public void publishConfig(String dataId, Object data) {
> >     LOG.warn("Config upload not support yet, please upload it in polaris first");
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > So if we change on the admin dashboard only, it can not sync to Polaris and the bootstrap application.
> 
> Now, Polaris-java SDK hasn't supported the sync config. But there is the other way to sync. Refer to [config sync codes](https://github.com/polarismesh/polaris-sync/blob/main/polaris-sync-config-plugins/config-polaris/src/main/java/cn/polarismesh/polaris/sync/config/plugins/polaris/PolarisRestUtils.java)

Thanks for that, let me see what I can do.💪

File: shenyu-bootstrap/pom.xml [hunk 1]
```diff
@@ -432,50 +432,58 @@
...
-                    <artifactId>grpc-grpclb</artifactId>
-                </exclusion>
-                <exclusion>
-                    <groupId>io.grpc</groupId>
-                    <artifactId>grpc-netty</artifactId>
-                </exclusion>
-            </exclusions>
-        </dependency>
+<!--        <dependency>-->
+<!--            <groupId>org.apache.shenyu</groupId>-->
```
Comment (damonxue) [inline shenyu-bootstrap/pom.xml:457]: I think it's no need to comment the dependencies.

File: shenyu-admin/pom.xml [hunk 2]
```diff
@@ -160,6 +161,12 @@
             <artifactId>nacos-client</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>com.tencent.polaris</groupId>
+            <artifactId>polaris-all</artifactId>
+            <version>1.12.0-SNAPSHOT</version>
```
Comment (damonxue) [inline shenyu-admin/pom.xml:167]: Maybe we can wait for the next release-version.

**Comment by fabian4** [comment]
Since the polaris has support client access for config create update and publish, this pr will be finished soon.

https://github.com/polarismesh/polaris/issues/1134

I am sorry for the delay in the project again.

**Review by dragon-zhang — CHANGES_REQUESTED** [review]
wait `com.tencent.polaris:polaris-all` have a release version

**Comment by fabian4** [comment]
> wait `com.tencent.polaris:polaris-all` have a release version

It is officially released now.
https://github.com/polarismesh/polaris-java/releases

File: shenyu-sync-data-center/shenyu-sync-data-polaris/pom.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        <dependency>
+            <groupId>org.apache.shenyu</groupId>
+            <artifactId>shenyu-sync-data-api</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.tencent.polaris</groupId>
+            <artifactId>polaris-all</artifactId>
+            <version>1.13.0-SNAPSHOT</version>
```
Comment (dragon-zhang) [inline shenyu-sync-data-center/shenyu-sync-data-polaris/pom.xml:52]: use release version

**Comment by dragon-zhang** [comment]
The rest parts looks fine.

**Review by dragon-zhang — APPROVED** [review]

**Review by damonxue — COMMENTED** [review]
pls @mahaitao617 review.

**Pull request merged**